.github/workflows/build.yml

```diff
@@ -0,0 +1,65 @@
+# This workflow will run:
+# 1. install Python dependencies
+# 2. build python package
+# 3. run unit tests
+# 4. Build documentation
+
+
+name: Python Package
+
+on:
+  push:
+    branches: [ "main" ]
+  pull_request:
+    branches: [ "main" ]
+
+jobs:
+  build_and_test:
+
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        os: ['ubuntu-latest', 'macos-latest', 'windows-latest']
+        python-version: ["3.10", "3.11", "3.12"]
+
+    steps:
+    - uses: actions/checkout@v3
+    - name: Set up Python ${{ matrix.python-version }}
+      uses: actions/setup-python@v3
+      with:
+        python-version: ${{ matrix.python-version }}
+    - name: Install dependencies
+      run: |
+        python -m pip install --upgrade pip
+        python -m pip install build
+        python -m pip install .[dev]
+    - name: Build
+      run: |
+        python -m build
+    - name: Unit tests
+      run: | 
+        pytest tests/
+
+  build_doc: 
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        os: ['ubuntu-latest', 'macos-latest', 'windows-latest']
```

Comment:
For testing the documentation, one `os` e.g. `ubuntu-latest` should be sufficient, we can remove the rest to make it concise and fast.